dashboard/lib/mega_section.rb

```diff
@@ -81,49 +90,75 @@ def self.create_teacher(email, password, name)
   #    results. (CSF allows imperfect results, CSD and CSP do not.)
   def self.create_section(options)
     script = Script.get_from_cache(options[:script_name])
+    script_levels = script.script_levels.includes(:levels)
 
     # Create the section
     section = create :section, script: script,
       **options.slice(:teacher, :name, :login_type, :grade)
 
-    current_student = 0
-
-    # Create students in section
-    (1..options[:num_students]).each do
+    # Create students
+    students = []
+    (1..options[:num_students]).each do |i|
       # Choose random properties and create student
       age_min = options[:age_min]
       age_max_inclusive = options[:age_max_inclusive]
-      current_student += 1
 
-      name = format(SAMPLE_STUDENT_NAME_FORMAT, current_student)
-      age = rng.rand(age_min..age_max_inclusive)
-      student_user = create :student, name: name, age: age
+      # We give all sample students the same parent email to make it easy to look them
+      # up after we bulk-create them here.
+      students << build(:student,
+        name: format(SAMPLE_STUDENT_NAME_FORMAT, i),
+        age: rng.rand(age_min..age_max_inclusive),
+        email: "mega_student_#{i}@example.com",
+        parent_email: SAMPLE_STUDENT_PARENT_EMAIL
+      )
+    end
+    User.import! students
 
+    students = User.where(parent_email: SAMPLE_STUDENT_PARENT_EMAIL)
```

Comment:
One annoying thing about activerecord-import with MySQL is that it can be tricky to get the ids of the rows you just inserted.  (Postgresql doesn't have this problem. :man_shrugging:)  Here, I'm using `parent_email` as a hack to quickly detect the rows we just added; `users` has an index of `parent_email` and by giving all the mega section students the same parent email we can look them up quickly after we create them, for use in the rest of the script.

Reply:
We could probably optimize this further by only retrieving ids `.select(:id)` since that's really all we need to create everything else.